qemu/tests/usb_common.py

```diff
@@ -0,0 +1,124 @@
+import json
+
+from collections import OrderedDict, Counter
+
+from virttest import utils_misc
+
+
+def parse_usb_topology(params):
+    """
+    Parse the usb devices topology to the params.
+
+    :param params: Dictionary with the test parameters.
+    :return: A list of dictionary ({usb_type_d0: usb_type}) of specified
+             usb topology.
+    """
+    # init number of usb devices based on device_num
+    params["usb_devices"] = ""
+    for i in xrange(int(params["device_num"])):
```

Comment:
Seems `usb_topology` contain the numbers, so what is the reason to introduce this param?

Reply:
Because in this case there are 5 devices they are all hubs.
In other cases, there could be 5 cases, 1 hub 2 mouses and 2 tablets.

Reply:
If so, `usb_topology` can describe this:
```python
usb_topology = {
    "usb-hub": 5,
    "usb-kbd": 2,
    ...
}
```
even more complex:
```python
usb_topology = {
    "usb-hub": {
        "usb-kbd": 2
    },
    "usb-hub": {
        "usb-mouse": 1,
        "usb-storage": 1
    },
    ...
}
```